Source/Charts/Data/Implementations/Standard/ChartData.swift

```diff
@@ -345,7 +389,7 @@ open class ChartData: NSObject, ExpressibleByArrayLiteral
     /// If set to true, this means that values can be highlighted programmatically or by touch gesture.
     @objc open var isHighlightEnabled: Bool
     {
-        get { return first { $0.highlightEnabled == false } == nil }
+        get { return allSatisfy { $0.isHighlightEnabled } }
```

Comment:
why remove == nil here?

Reply:
`first(where:)` returns nil if it doesn't find an object satisfying the condition. In the old case, we checked that they were all false. If one was true, then `first(where:)` would return an object.

`allSatisfy` returns a boolean. `isHighlightEnabled == true` IFF all of the dataSets satisfy the condition `isHighlightEnabled == true`